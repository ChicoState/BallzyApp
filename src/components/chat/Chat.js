import React from 'react';

import {
  Text,
  View,
  StyleSheet,
  Image,
  TextInput,
  TouchableOpacity
} from 'react-native';


//import Messaging from './Messaging';


class Chat extends React.Component {
  static navigationOptions = {
    // Note: By default the icon is only shown on iOS. Search the showIcon option below.
    title: 'Chat',
    tabBarIcon: () => (
      <Image
        source={require('../navigationbar/conversations48x48.png')}
        style = {{width: 32, height: 32}}
      />
    ),
  };

  state = {
    name: '',
  };

  render() {
    return(
      <View
        style={styles.container}
      >
        <View style={styles.loginContainer}>
          <Text style={styles.title}>
            Enter user nickname!
          </Text>
          <TextInput
            style = {styles.nameInput}
            placeholder='ex: jdoe530'
            onChangeText={ (text) => { this.setState({name: text}); } }
            value = {this.state.name}
          />
          <TouchableOpacity
            style = {styles.button}
            onPress = {() => this.props.navigation.navigate('Messaging')}
          >
            <Text style = {styles.buttonText}>
              Next
            </Text>
          </TouchableOpacity>
        </View>
      </View>
    );
  }

}
//this a custom prop
var styles = StyleSheet.create( {
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#dcebfc',
  },
  loginContainer: {
    justifyContent: 'center',
    alignItems: 'center',
    height: 270,
    width: 300,
    backgroundColor: '#42033D',
    borderColor: '#ffe500',
    borderWidth: 3,
    borderRadius: 5,
  },
  title: {
    alignItems: 'center',
    justifyContent: 'center',
    fontSize: 20,
    fontWeight: '600',
    color: 'white',
    width: 250,
  },
  nameInput: {
    padding: 5,
    width: 250,
    height: 40,
    borderColor: 'black',
    margin: 20,
    backgroundColor: 'white'
  },
  button: {
    height: 50,
    width: 250,
    borderWidth: 0,
    borderRadius: 5,
    padding: 10,
    marginTop: 5,
    backgroundColor: '#4392F1'
  },
  buttonText: {
    fontSize: 20,
    fontWeight: '600',
    width: 50,
    alignSelf: 'center',
    textAlign: 'center',
    color: '#FFFFFF'
  }
} );

export default Chat;
