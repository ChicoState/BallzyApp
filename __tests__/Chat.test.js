'use strict';

import 'react-native';
import React from 'react';
//import React from 'react';
import Chat from '../src/components/chat/Chat';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Chat />
  ).toJSON();
  expect(tree).toMatchSnapshot();

});
