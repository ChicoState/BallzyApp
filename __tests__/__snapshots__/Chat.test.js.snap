exports[`test renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "#dcebfc",
      "flex": 1,
      "justifyContent": "center",
    }
  }>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 10,
        "justifyContent": "center",
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFFF31",
          "borderRadius": 5,
          "borderWidth": 1,
          "fontSize": 20,
          "fontWeight": "600",
          "justifyContent": "center",
          "width": 250,
        }
      }>
      Enter user nickname!
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="ex: jdoe530"
      style={
        Object {
          "backgroundColor": "white",
          "borderColor": "black",
          "height": 40,
          "margin": 20,
          "padding": 5,
          "width": 250,
        }
      }
      value="" />
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#4392F1",
          "borderRadius": 5,
          "borderWidth": 0,
          "height": 50,
          "marginTop": 5,
          "opacity": 1,
          "padding": 10,
          "width": 250,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "color": "#FFFFFF",
            "fontSize": 20,
            "fontWeight": "600",
            "textAlign": "center",
            "width": 50,
          }
        }>
        Next
      </Text>
    </View>
  </View>
</View>
`;
